lib/builders/gpu-builder.ts

```diff
@@ -0,0 +1,133 @@
+import * as blueprints from '../../lib';
+import * as utils from "../utils";
+import { NestedStack, NestedStackProps } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as eks from "aws-cdk-lib/aws-eks";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { NodegroupAmiType } from 'aws-cdk-lib/aws-eks';
+
+/**
+ * Configuration options for GPU Builder.
+ */
+export interface GpuOptions {
+    /**
+     * Required, Kubernetes version to use for the cluster.
+     */
+    kubernetesVersion: eks.KubernetesVersion,
+    /** 
+     * Required, Instance class to use for the cluster. 
+     */
+    instanceClass: ec2.InstanceClass,
+    /** 
+     * Required, Instance size to use for the cluster. 
+     */
+    instanceSize: ec2.InstanceSize,
+    /** 
+     * Optional, AMI Type for GPU Nodes @ default AL2_X86_64_GPU 
+     */
+    gpuAmiType?: NodegroupAmiType,
+    /** 
+     * Optional, Desired number of nodes to use for the cluster. 
+     */
+    desiredNodeSize?: number,
+    /** 
+     * Optional, Minimum number of nodes to use for the cluster. 
+     */
+    minNodeSize?: number,
+    /** 
+     * Optional, Maximum number of nodes to use for the cluster. 
+     */
+    maxNodeSize?: number,
+    /** 
+     * Optional, Block device size.
+     */
+    blockDeviceSize?: number,
+    /**
+     * Optional, Cluster Provider Tags.
+     */
+    clusterProviderTags?: {
+        [key: string]: string;
+    },
+
+    /**
+     * Optional, Node Group Tags for AL2 nodes 
+     * which run standard cluster software.
+     */
+    nodeGroupTags?: {
+        [key: string]: string;
+    }
+}
+
+/**
+ * Default props to be used when creating the GPU nodes 
+ * for EKS cluster
+ */
+const defaultOptions: GpuOptions = {
+    kubernetesVersion: eks.KubernetesVersion.of("1.27"),
+    instanceClass: ec2.InstanceClass.G5,
+    instanceSize: ec2.InstanceSize.XLARGE,
+    gpuAmiType: NodegroupAmiType.AL2_X86_64_GPU,
+    desiredNodeSize: 2,
+    minNodeSize: 2,
+    maxNodeSize: 3,
+    blockDeviceSize: 50,
+    clusterProviderTags: {
+        "Name": "blueprints-gpu-eks-cluster",
+        "Type": "generic-gpu-cluster"
+    },
+    nodeGroupTags: {
+        "Name": "Mng-linux",
+        "Type": "Managed-linux-Node-Group",
+        "LaunchTemplate": "Linux-Launch-Template",
+    }
+  };
+export class GpuBuilder extends blueprints.BlueprintBuilder {
+    /**
+     * This method helps you prepare a blueprint for setting up observability 
+     * returning an array of blueprint addons for AWS managed open source services
+     */
+    public enableGpu(values?: blueprints.Values): GpuBuilder {
```

Comment:
what are these values? it is going confusing to the customers which values are allowed to be passed. the way it stands, it is just key value pairs and no indication of the schema. Did you mean to allow passing some particular type here?

Reply:
Added valueSchema similar to container insights values